src/components/member-entries/MemberEntries.tsx

```diff
@@ -23,13 +24,28 @@ export const MemberEntries = (props: MemberEntriesProps) => {
     setTeamMembers(props.teamMembers);
   }, []);
 
+  const [addTeamMember, { data }] = useMutation(CREATE_TEAM_MEMBER);
+
   async function handleClick(newMemberEntry: Types.MemberEntryProps) {
-    const formattedNewEntry = await addMemberEntry(newMemberEntry);
-    if (formattedNewEntry) {
-      setTeamMembers([...teamMembers, formattedNewEntry]);
-    }
+    addTeamMember({
+      variables: {
+        firstName: newMemberEntry.firstName,
+        lastName: newMemberEntry.lastName,
+        emailAddress: newMemberEntry.emailAddress,
+        label: newMemberEntry.label,
+        client: newMemberEntry.client,
+        role: newMemberEntry.role,
+        startingDate: newMemberEntry.startingDate,
+      },
+    });
   }
 
+  useEffect(() => {
+    if (data) {
+      setTeamMembers([...teamMembers, data.createTeamMember]);
+    }
+  }, [data]);
```

Comment:
While a nice solution, use the `onCompleted` callback that Apollo offers. It's easier to read and forces you to understand the library you're using better